gateway-ha/src/main/java/io/trino/gateway/ha/security/LbOAuthManager.java

```diff
@@ -72,35 +79,32 @@ public String getUserIdField()
      */
     public Response exchangeCodeForToken(String code, String redirectLocation)
     {
-        String tokenEndpoint = oauthConfig.getTokenEndpoint();
-        String clientId = oauthConfig.getClientId();
-        String clientSecret = oauthConfig.getClientSecret();
-        String redirectUri = oauthConfig.getRedirectUrl();
-        String redirectWebUrl = oauthConfig.getRedirectWebUrl();
-        Client oauthClient = ClientBuilder.newBuilder().build();
+        TokenRequest tokenRequest = new TokenRequest(
+                oauthConfig.getTokenEndpoint(),
+                new ClientSecretBasic(new ClientID(oauthConfig.getClientId()), new Secret(oauthConfig.getClientSecret())),
+                new AuthorizationCodeGrant(new AuthorizationCode(code), oauthConfig.getRedirectUrl()));
 
-        Form form = new Form().param("grant_type", "authorization_code")
-                .param("client_id", clientId)
-                .param("client_secret", clientSecret)
-                .param("code", code)
-                .param("redirect_uri", redirectUri);
-
-        Response tokenResponse = oauthClient
-                .target(tokenEndpoint)
-                .request()
-                .post(Entity.form(form));
-
-        if (tokenResponse.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
-            String message = format("token response failed with code %d - %s", tokenResponse.getStatus(), tokenResponse.readEntity(String.class));
+        TokenResponse tokenResponse;
+        try {
+            tokenResponse = OIDCTokenResponseParser.parse(tokenRequest.toHTTPRequest().send());
+        }
+        catch (ParseException | IOException e) {
+            String message = format("Failed to parse token response: %s", e.getMessage());
             log.error(message);
-            return Response.status(500).entity(message).build();
+            return Response.status(401).build();
```

Comment:
Please consider using `jakarta.ws.rs.core.Response.Status#UNAUTHORIZED` instead. Same for another place.